### PR TITLE
Add REST endpoint for tokens

### PR DESCRIPTION
**Pull request by Mykhailenko**

### CODE REVIEW

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java [hunk 1]
```diff
@@ -3096,4 +3097,23 @@ private boolean nodesRecoveryEnabledForNode(RMNode rmNode) {
...
+        }
+    }
+
+    @Override
+    public void removeNodeToken(String nodeUrl, String token) throws RMException {
+        if (allNodes.containsKey(nodeUrl)) {
+            RMNode rmNode = allNodes.get(nodeUrl);
+            rmNode.removeToken(token);
+        } else {
+            throw new RMException("Unkonwn node " + nodeUrl);
```
Comment (MaelAudren) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java]: Unkonwn => Unknown

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java [hunk 2]
```diff
@@ -3096,4 +3097,23 @@ private boolean nodesRecoveryEnabledForNode(RMNode rmNode) {
         return isNodesRecoveryEnabled() && rmNode.getNodeSource().nodesRecoverable();
     }
 
+    @Override
+    public void addNodeToken(String nodeUrl, String token) throws RMException {
+        if (allNodes.containsKey(nodeUrl)) {
+            RMNode rmNode = allNodes.get(nodeUrl);
+            rmNode.addToken(token);
+        } else {
+            throw new RMException("Unkonwn node " + nodeUrl);
```
Comment (MaelAudren) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java]: Unkonwn => Unknown

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/rmnode/RMNodeImpl.java [hunk 3]
```diff
@@ -492,4 +493,23 @@ public boolean isDeploying() {
         return false;
     }
 
+    @Override
+    public void addToken(String token) {
+        PrincipalPermission principalPermission = (PrincipalPermission) getUserPermission();
+        principalPermission.addPermission(new TokenPrincipal(token));
```
Comment (MaelAudren) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/rmnode/RMNodeImpl.java]: I don't understand where the local variable `principalPermission` is used ?

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java (hunk 1 above)
Comment (Mykhailenko) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java]: thanks

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/rmnode/RMNodeImpl.java (hunk 3 above)
Comment (Mykhailenko) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/rmnode/RMNodeImpl.java]: I add to the `principalPermission` new permission

**Pull request merged**